pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -256,7 +256,7 @@ def get_evaluate_epoch_results(self, test_mode):
             print('-' * 80)
             for result_idx, results in enumerate(self.eval_loop_results):
                 print(f'DATALOADER:{result_idx} TEST RESULTS')
-                pprint(results)
+                pprint({k: v.tolist() if isinstance(v, torch.Tensor) else v for k, v in results.items()})
```

Comment:
Calling `.tolist()` will result in scalar tensors being wrapped in a list:
{'test_loss': [1.1703]}
Instead we should probably call `item` for scalar tensors and `tolist` for non-scalar tensors:
```suggestion
                convert = lambda x: x.item() if x.numel()==1 else x.tolist()
                pprint({k: convert(v) if isinstance(v, torch.Tensor) else v for k, v in results.items()})
```

Reply:
I had actually checked for that and at least in PyTorch 1.7.1 `torch.tensor(1.1703).tolist()` doesn't return square brackets to me: `{'test_loss': 1.1703}`. What PyTorch version are you using?

Reply:
I am running torch v1.6. Can we do something to be backward compatible to earlier versions of pytorch (since lightning in general is backward compatible to pytorch v1.3)?

Reply:
@SkafteNicki I've tried with PyTorch 1.6 and 1.3 as well, and none of them return the square brackets. Can you please confirm `torch.tensor(1.1703).tolist()` gives you square brackets?

Reply:
I commented because this simple example will fail:
```torch.randn(1).tolist() 
>>> [0.2993074953556061]
``` 
Maybe it is to artificial, but it seems that `tolist` only works when the shape of the tensor is `torch.Size([])` and not in general when it is a 1 element tensor.